CMakeLists.txt

```diff
@@ -7,18 +7,61 @@ project(libswid)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
-file(GLOB_RECURSE sources      src/*.cpp src/*.h)
+list(APPEND private_sources
+	${CMAKE_CURRENT_SOURCE_DIR}/src/loader-generic.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/loader-generic.h
+	${CMAKE_CURRENT_SOURCE_DIR}/src/loader-generic-notemplate.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/src/libswid.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/src/libswid-base.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/src/SWIDStruct.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/SWIDStruct.h
+	${CMAKE_CURRENT_SOURCE_DIR}/src/Translator.cpp ${CMAKE_CURRENT_SOURCE_DIR}/src/Translator.h
+)
+
+# We can't include extension-less files here, because they puzzle the coverage support:
+# -> no append of ${CMAKE_CURRENT_SOURCE_DIR}/src/libswid
+list(APPEND public_sources
+	${CMAKE_CURRENT_SOURCE_DIR}/src/libswid.h
+	${CMAKE_CURRENT_SOURCE_DIR}/src/libswid-base.h
+	${CMAKE_CURRENT_SOURCE_DIR}/src/swid-common.h
+)
+
 file(GLOB_RECURSE sources_test_cxx test/unit/*.cpp)
 file(GLOB_RECURSE sources_test_c test/unit/*.c)
 # You can use set(sources src/main.cpp) etc if you don't want to
 # use globbing to find files automatically.
 
-find_package(XercesC)
-find_package(TinyXML)
+option(BUILD_XERCES_BACKEND "Whether to build Xerces-C backend" ON)
+option(BUILD_TINYXML_BACKEND "Whether to build TinyXML backend" ON)
+
+if (BUILD_XERCES_BACKEND)
+	find_package(XercesC)
```

Comment:
so this is a personal opinion but I think the other approach is better - always look for all packages, if they are not found disable building backends. DOn't ask users to know what they ahve installed, tell them

Reply:
plus you can refactor both conditions the checks are dupes

Reply:
I don't get what you mean - my intention is to build all backends by default. If a required library is missing, the backend won't be built.
I got the point about feedback and refactoring though.

Reply:
most projects look for all dependencies, optional AND required, all of them. then based on what was found they will turn building some parts of the project on and off and user can override. you do the opposite, you let the user choose what to build and then you look for packages.

Reply:
Tell me what you think of the recent modification.